views/models/UserConfigurationsModel/index.js

```diff
@@ -0,0 +1,71 @@
+import { apiRoutes } from '../../routes/routes'
+import { BASE_REQUEST_OPTIONS } from '../../../constants'
+
+const UserConfigurationsModel = {
+  all: async (userId) => {
+    try {
+      const response = await fetch(
+        apiRoutes.configurations.userConfigurations(userId),
+        {
+          ...BASE_REQUEST_OPTIONS,
+        }
+      )
+      if (response.status !== 200) throw new Error(response.error)
+
+      return response.json()
+    } catch (error) {
+      throw new Error(error)
+    }
+  },
+  create: async (userId, configAttributes) => {
+    try {
+      const response = await fetch(
+        apiRoutes.configurations.userConfigurations(userId),
+        {
+          ...BASE_REQUEST_OPTIONS,
+          method: 'POST',
+          body: JSON.stringify(configAttributes),
+        }
+      )
+      const status = response.status
+
+      return { status, ...response.json() }
+    } catch (error) {
+      throw new Error(error)
+    }
+  },
+  destroy: async (userId, configId) => {
+    try {
+      const response = await fetch(
+        apiRoutes.configurations.userConfiguration(userId, configId),
+        {
+          ...BASE_REQUEST_OPTIONS,
+          method: 'DELETE',
+        }
+      )
+
+      return response
```

Comment:
should this return an object with a status and data key like the others?

Reply:
the controller only returns a status

Reply:
yeah but we want consistency in the response objects.

Reply:
I still think we need consistency in our response objects. In this case we are returning a response object, in orders we are returning the response json. What about something like

```ts
{ status: number, data?: object, error?: string }
```

this would mean that here we would `return { status: response.status }` and in the other methods we would do

```js
let response = { status: 500 }

try {
  response = await fetch(...)
  const { data, error } = await response.json()

  return { status: response.status, data, error }
} catch(e) {
  return { response.status, error: e.message }
}
```

Reply:
[22063e9](https://github.com/AMP-SCZ/dpdash/pull/310/commits/22063e9b9ef9c557741703dee541a7e18fe68e26)